tendermint/src/time.rs

```diff
@@ -106,33 +114,35 @@ impl From<Time> for DateTime<Utc> {
     }
 }
 
-impl From<SystemTime> for Time {
-    fn from(t: SystemTime) -> Time {
-        Time(t.into())
-    }
-}
-
-impl From<Time> for SystemTime {
-    fn from(t: Time) -> SystemTime {
-        t.0.into()
-    }
-}
-
 impl Add<Duration> for Time {
-    type Output = Self;
+    type Output = Result<Self, Error>;
 
     fn add(self, rhs: Duration) -> Self::Output {
-        let st: SystemTime = self.into();
```

Comment:
I remove the use of `SystemTime` for addition/substration here to support no_std. To perform safe addition in `chrono`, we have to take care of integer overflows by returning error. This is important as the overflow result may cause computation such as trusted period to return incorrect result.